core/ir/arm/instr.c

```diff
@@ -39,19 +39,17 @@
 bool
 instr_set_isa_mode(instr_t *instr, dr_isa_mode_t mode)
 {
-    if (mode == DR_ISA_ARM_THUMB)
-        instr->flags |= INSTR_THUMB_MODE;
-    else if (mode == DR_ISA_ARM_A32)
-        instr->flags &= ~INSTR_THUMB_MODE;
-    else
+    if ((mode != DR_ISA_ARM_THUMB) && (mode != DR_ISA_ARM_A32)) {
         return false;
+    }
+    instr->isa_mode = mode;
     return true;
 }
 
 dr_isa_mode_t
 instr_get_isa_mode(instr_t *instr)
 {
-    return TEST(INSTR_THUMB_MODE, instr->flags) ? DR_ISA_ARM_THUMB : DR_ISA_ARM_A32;
+    return instr->isa_mode;
```

Comment:
Let's move this function to instr_shared.c if all arches are now the same.

Reply:
Unfortunately, some instr_get_isa_mode() were just returning a specific ISA mode (without retrieving it from the instr_t flags).
Meaning that they were able to return a valid ISA mode, even though it was not previously set in the flags of instr_t.
There were several (>300) tests relying on the fact that the ISA mode could still be retrieved without being set first, and I wasn't sure if that was intentional, so I modified the instr_get_isa_mode() to reflect the previous behavior.
This also means that they are not all the same anymore.

Reply:
I think we just need instr_create() to set the mode for A64 and RISCV?

Reply:
You're right, that fixed it.
Now that all instr_get_isa_mode() look the same I moved them to core/ir/instr_shared.c